core/src/test/groovy/com/netflix/conductor/core/execution/tasks/StartWorkflowSpec.groovy

```diff
@@ -0,0 +1,119 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.netflix.conductor.core.execution.tasks
+
+
+import javax.validation.ConstraintViolation
+import javax.validation.Validator
+
+import com.netflix.conductor.common.config.ObjectMapperProvider
+import com.netflix.conductor.core.exception.ApplicationException
+import com.netflix.conductor.core.execution.WorkflowExecutor
+import com.netflix.conductor.model.TaskModel
+import com.netflix.conductor.model.WorkflowModel
+
+import spock.lang.Specification
+import spock.lang.Subject
+
+import static com.netflix.conductor.core.execution.tasks.StartWorkflow.START_WORKFLOW_PARAMETER
+import static com.netflix.conductor.model.TaskModel.Status.FAILED
+import static com.netflix.conductor.model.TaskModel.Status.SCHEDULED
+
+/**
+ * Unit test for StartWorkflow. Success and Javax validation cases are covered by the StartWorkflowSpec in test-harness module.
+ */
+class StartWorkflowSpec extends Specification {
+
+    @Subject
+    StartWorkflow startWorkflow
+
+    WorkflowExecutor workflowExecutor
+    Validator validator
+    WorkflowModel workflowModel
+    TaskModel taskModel
+
+    def setup() {
+        workflowExecutor = Mock(WorkflowExecutor.class)
+        validator = Mock(Validator.class) {
+            validate(_) >> new HashSet<ConstraintViolation<Object>>()
+        }
+
+        def inputData = [:]
+        inputData[START_WORKFLOW_PARAMETER] = ['name': 'some_workflow']
+        taskModel = new TaskModel(status: SCHEDULED, inputData: inputData)
+        workflowModel = new WorkflowModel()
+
+        startWorkflow = new StartWorkflow(new ObjectMapperProvider().getObjectMapper(), validator)
+    }
+
+    def "StartWorkflow task is asynchronous"() {
+        when:
+        def async = startWorkflow.isAsync()
+
+        then:
+        async
+    }
+
+    def "start_workflow parameter is missing"() {
+        given: "a task with no start_workflow in input"
+        taskModel.inputData = [:]
+
+        when:
+        startWorkflow.start(workflowModel, taskModel, workflowExecutor)
+
+        then:
+        taskModel.status == FAILED
+        taskModel.reasonForIncompletion != null
+    }
+
+    def "ObjectMapper throws an IllegalArgumentException"() {
+        given: "a task with no start_workflow in input"
+        taskModel.inputData[START_WORKFLOW_PARAMETER] = "I can't be converted to StartWorkflowRequest"
+
+        when:
+        startWorkflow.start(workflowModel, taskModel, workflowExecutor)
+
+        then:
+        taskModel.status == FAILED
+        taskModel.reasonForIncompletion != null
+    }
+
+    def "WorkflowExecutor throws a retryable exception"() {
+        when:
+        startWorkflow.start(workflowModel, taskModel, workflowExecutor)
+
+        then:
+        taskModel.status == SCHEDULED
```

Comment:
please add assert on `reasonForIncompletion`

Reply:
The `reasonForIncompletion` is not set in this case. Add an assert for `reasonForIncompletion == null`

Reply:
ah yes, you're right, misread it